samples/subsys/usb/uac2_implicit_feedback/src/main.c

```diff
@@ -0,0 +1,570 @@
+/*
+ * Copyright (c) 2023-2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <sample_usbd.h>
+#include "feedback.h"
+
+#include <zephyr/cache.h>
+#include <zephyr/device.h>
+#include <zephyr/usb/usbd.h>
+#include <zephyr/usb/class/usbd_uac2.h>
+#include <zephyr/drivers/i2s.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(uac2_sample, LOG_LEVEL_INF);
+
+#define HEADPHONES_OUT_TERMINAL_ID UAC2_ENTITY_ID(DT_NODELABEL(out_terminal))
+#define MICROPHONE_IN_TERMINAL_ID UAC2_ENTITY_ID(DT_NODELABEL(in_terminal))
+
+#define SAMPLES_PER_SOF     48
+#define SAMPLE_FREQUENCY    (SAMPLES_PER_SOF * 1000)
+#define SAMPLE_BIT_WIDTH    16
+#define NUMBER_OF_CHANNELS  2
+#define BYTES_PER_SAMPLE    DIV_ROUND_UP(SAMPLE_BIT_WIDTH, 8)
+#define BYTES_PER_SLOT      (BYTES_PER_SAMPLE * NUMBER_OF_CHANNELS)
+#define MIN_BLOCK_SIZE      ((SAMPLES_PER_SOF - 1) * BYTES_PER_SLOT)
+#define BLOCK_SIZE          (SAMPLES_PER_SOF * BYTES_PER_SLOT)
+#define MAX_BLOCK_SIZE      ((SAMPLES_PER_SOF + 1) * BYTES_PER_SLOT)
+
+/* Absolute minimum is 5 TX buffers (1 actively consumed by I2S, 2nd queued as
+ * next buffer, 3rd acquired by USB stack to receive data to, and 2 to handle
+ * SOF/I2S offset errors), but add 2 additional buffers to prevent out of memory
+ * errors when USB host decides to perform rapid terminal enable/disable cycles.
+ */
+#define I2S_BLOCKS          7
+K_MEM_SLAB_DEFINE_STATIC(i2s_tx_slab, MAX_BLOCK_SIZE, I2S_BLOCKS, 4);
+K_MEM_SLAB_DEFINE_STATIC(i2s_rx_slab, MAX_BLOCK_SIZE, I2S_BLOCKS, 4);
+
+struct usb_i2s_ctx {
+	const struct device *i2s_dev;
+	bool headphones_enabled;
```

Comment:
Would it be more descriptive to name this playback and recording? Or out and in?